ecosystem/indexer-grpc/indexer-grpc-in-memory-cache-benchmark/src/main.rs

```diff
@@ -0,0 +1,210 @@
+            ))
+        })
+        .collect();
+    let redis_resp = redis::Value::Bulk(values);
+    MockCmd::new(redis::cmd("MGET").arg(keys), Ok(redis_resp))
+}
+
+fn generate_redis_commands(transaction_size: u64) -> Vec<MockCmd> {
+    let mut redis_commands = vec![
+        // No warm up.
+        mock_get_latest_version(0),
+        // Wait for a little bit since we're spawning up clients.
+        mock_get_latest_version(0),
+        mock_get_latest_version(0),
+        mock_get_latest_version(0),
+        mock_get_latest_version(0),
+        mock_get_latest_version(0),
+    ];
+
+    // number of batches.
+    let num_batches = 100 * 1024 * 1024 / (transaction_size as usize);
+    // Start to update.
+    for i in 0..num_batches {
+        redis_commands.push(mock_get_latest_version(
+            (i as i64 + 1) * MAX_REDIS_FETCH_BATCH_SIZE as i64,
+        ));
+        redis_commands.push(mock_mget_transactions(
+            i as u64 * (MAX_REDIS_FETCH_BATCH_SIZE as u64),
+            MAX_REDIS_FETCH_BATCH_SIZE,
+            transaction_size as usize,
+        ));
+    }
+
+    redis_commands
+}
+
+// Creates a mock Redis connection with a stream of commands.
+// The pattern is:
+// 1. Warm up: get latest_version, get transactions.
+// 2. loop of fetching the data
+fn create_mock_redis(transaction_size: u64) -> MockRedisConnection {
+    MockRedisConnection::new(generate_redis_commands(transaction_size))
+}
+
+// Spawn a task that consumes the cache. Return tps as a result.
+fn spawn_cache_consumption_task(
+    cache: Arc<InMemoryCache>,
+    duration_in_secs: u64,
+) -> tokio::task::JoinHandle<f64> {
+    tokio::spawn(async move {
+        let start = std::time::Instant::now();
+        let mut total_consumed_transactions = 0;
+        let mut current_version = 0;
+        while start.elapsed().as_secs() < duration_in_secs {
+            let txns = cache.get_transactions(current_version).await;
+            for txn in txns {
+                assert_eq!(txn.version, current_version);
+                total_consumed_transactions += 1;
+                current_version = txn.version + 1;
+            }
+            // Dropped here.
+        }
+        total_consumed_transactions as f64 / start.elapsed().as_secs_f64()
+    })
+}
+
+async fn test_small_transactions() {
+    let transaction_size = 2 * 1024;
+    let task_count = 10;
+    let redis_connection = create_mock_redis(transaction_size);
+    let cache = Arc::new(
+        InMemoryCache::new_with_redis_connection(
+            redis_connection,
+            StorageFormat::Lz4CompressedProto,
+        )
+        .await
+        .unwrap(),
+    );
+    let tasks = (0..task_count)
+        .map(|_| spawn_cache_consumption_task(cache.clone(), 20))
+        .collect::<Vec<_>>();
+    // join all the tasks.
+    let tps = futures::future::join_all(tasks)
+        .await
+        .iter()
+        .map(|r| r.as_ref().unwrap())
+        .sum::<f64>()
+        / task_count as f64;
```

Comment:
nit: maybe we can parametrize transaction size and others, so we can add more test scenarios if needed by reducing duplication. 

```
#[tokio::main(flavor = "multi_thread", worker_threads = 16)]
async fn main() -> anyhow::Result<()> {
    run_transaction_test(2 * 1024, 10, 20, 20_000.0).await;
    run_transaction_test(2 * 1024, 1000, 20, 100.0).await;
    run_transaction_test(1024 * 1024, 20, 5, 500.0).await;
    Ok(())

async fn run_transaction_test(transaction_size: usize, task_count: usize, duration: u64, expected_tps: f64) {
}
```

Reply:
good call; done